waspc/src/Parser/NpmDependencies.hs

```diff
@@ -0,0 +1,31 @@
+module Parser.NpmDependencies
+    ( npmDependencies
+    ) where
+
+import qualified Data.Aeson                as Aeson
+import qualified Data.ByteString.Lazy.UTF8 as BLU
+import qualified Data.HashMap.Strict       as M
+import           Text.Parsec               (try)
+import           Text.Parsec.String        (Parser)
+
+import qualified Lexer                     as L
+import qualified NpmDependency             as ND
+import qualified Parser.Common             as P
+import           Wasp.NpmDependencies      (NpmDependencies)
+import qualified Wasp.NpmDependencies      as NpmDependencies
+
+
+npmDependencies :: Parser NpmDependencies
+npmDependencies = try $ do
+    L.reserved L.reservedNameDependencies
+    closureContent <- P.waspNamedClosure "json"
+    let jsonBytestring = BLU.fromString $ "{ " ++ closureContent ++ " }"
+    npmDeps <- case Aeson.eitherDecode' jsonBytestring :: Either String (M.HashMap String String) of
```

Comment:
Here return type (which I specified explicitely) decides how string will be decoded by Aeson -> meaning it is polymorphic in return type, pretty cool! This is also how Haskell Regex libraries work normally.